draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -86,22 +86,32 @@ with dynamically generated Multicast DNS (mDNS) names.
 Introduction {#problems}
 ============
 
-As detailed in {{IPHandling}}, exposing client private IP addresses by default
-maximizes the probability of successfully creating direct peer-to-peer
-connection between two clients, but creates a significant surface for user
-fingerprinting. {{IPHandling}} recognizes this issue, but also admits that there
-is no current solution to this problem; implementations that choose to use
-Mode 3 to address the privacy concerns often suffer from failing or suboptimal
-connections in WebRTC applications. This is particularly an issue on unmanaged
-networks, typically homes or small offices, where NAT loopback may not be
-supported.
-
-This document proposes an overall solution to this problem by registering
-ephemeral mDNS {{RFC6762}} names for each local private IP address, and then
-providing those names, rather than the IP addresses, to the web application
-when it gathers ICE candidates. WebRTC implementations resolve these names
-to IP addresses and perform ICE processing as usual, but the actual IP addresses
-are not exposed to the web application.
+As detailed in {{IPHandling}}, exposing client private IP addresses by
+default to web applications maximizes the probability of successfully creating
+direct peer-to-peer connections between clients, but creates a significant
+surface for user fingerprinting. {{IPHandling}} recognizes this issue, but also
+admits that there is no current solution to this problem; implementations that
+choose to use Mode 3 to address the privacy concerns often suffer from failing
+or suboptimal connections in WebRTC applications. This is particularly an issue
+on unmanaged networks, typically homes or small offices, where NAT loopback may
+not be supported.
+
+This document proposes an overall solution to this problem by providing a
+mechanism for WebRTC implementations to register
+ephemeral mDNS {{RFC6762}} names for local private IP addresses, and then
+provide those names, rather than the IP addresses, in the ICE candidates
+they surface to web applications.
+
+Upon receiving ICE candidates with mDNS names, remote clients
+will resolve these names to IP addresses and
+perform ICE processing as usual. In the case where the remote client is a web
+application, the WebRTC implementation will manage this resolution internally
+and will not disclose the actual IP addresses to the application, as above.
```

Comment:
'as above' is not very clear. Probably refers to mode 2, maybe we can do
s/, as above//